02_knight_and_princess/xieyu_wang/main.cpp

```diff
@@ -124,14 +106,16 @@ class Graph {
   }
 
  private:
-  int getNextEdgeId() {
-    vertexIdCounter_++;
-    return vertexIdCounter_ - 1;
-  }
-  int getNextVertexId() {
-    edgeIdCounter_++;
-    return edgeIdCounter_ - 1;
+  Vertex& getVertex(int id) const {
+    for (const auto& vertex : vertices_) {
+      if (vertex.id == id) {
+        return (Vertex&)vertex;
+      }
```

Comment:
Why do you need to use cast here?
```suggestion
    for (auto& vertex : vertices_) {
      if (vertex.id == id) {
        return vertex;
      }
```

Reply:
![Screenshot 2021-11-07 122617](https://user-images.githubusercontent.com/30003216/140632291-5a6ce8f2-6b21-4631-a425-395300065275.png)
There will be a error if i dont use (Vertex&) even if i removed const

Reply:
And what the error?

Reply:
And don't press `Resolved` if it is not reloved